src/Tasks/vmOperations/vmOperations.ts

```diff
@@ -58,7 +58,7 @@ export class VmOperations {
                 break;
             case "Take Snapshot of Virtual Machines":
                 var snapshotName = tl.getInput("snapshotName", true);
-                var snapshotVMMemory = "true";
+                var snapshotVMMemory = tl.getInput("snapshotMemory", true);

```

Comment:
It should be tl.getInput("snapshotMemory", false). Second parameter says whether it is mandatory or optional parameter